src/monitor/pgautofailover.sql

```diff
@@ -345,6 +348,33 @@ $$;
 grant execute on function pgautofailover.get_coordinator(text)
    to autoctl_node;
 
+
+CREATE FUNCTION pgautofailover.get_most_advanced_standby
+ (
+   IN formationid       text default 'default',
+   IN groupid           int default 0,
+   OUT node_id          bigint,
+   OUT node_name        text,
+   OUT node_host        text,
+   OUT node_port        int,
+   OUT node_lsn         pg_lsn,
+   OUT node_is_primary  bool
+ )
+RETURNS SETOF record LANGUAGE SQL STRICT
+AS $$
+   select nodeid, nodename, nodehost, nodeport, reportedlsn, false
+     from pgautofailover.node
+    where formationid = $1
```

Comment:
Shouldn't we add `health ` to the `WHERE`  clause instead of `ORDER BY`?

Reply:
The problem is that then we would have no result when there's no known-healthy node, and I would rather pick one and try to connect anyway. We might be lucky and connect before the monitor did change the health column.

That said, it's a little strange. Again, worth some testing and a chat, I believe.

Reply:
It'd be nice to  have this comment on the function as well, because it is not easy to understand for readers